### PR TITLE
Amr game stat class

### PR DESCRIPTION
**Pull request by AlexMR-93**
refactored class code. 
thinking about maybe having a module for all the helper methods to ry to get under 100 lines.

### CODE REVIEW

File: lib/game_stats.rb [hunk 1]
```diff
@@ -1,10 +1,116 @@
...
+    all_home_games = []
+
+    @game_teams.each do |row|
+      home_wins << row if row[:hoa] == "home" && row[:result] == "WIN"
+      all_home_games << row if row[:hoa] == "home"
+    end
+
+    ((home_wins.count / all_home_games.count.to_f).round(2))
+  end
+
```
Comment (gjcarew) [inline lib/game_stats.rb:23]: You could combine percentage_home_wins, percentage_visitor_wins, and percentage_ties into one helper method: 
`def percentage_result(num_condition, denom_condition)`
  `  num_count = 0`
  `  denom_count = 0`
  `  @game_teams.each do |row|`
    `    num_count += 1 if num_condition`
    `    denom_count += 1 if denom_condition`
  `  end`
  `  (num_count / denom_count.to_f).round(2)`
`end`

For example, the `percentage_home_wins` method could look like this:
`def percentage_home_wins`
  `  home_wins = row[:hoa] == "home" && row[:result] == "WIN"`
  `  all_home_games = row[:hoa] == "home"`
  `  percentage_result(home_wins, all_home_games)`
`end`

Edit: Tried it in Ruby and couldn't get it to work, so maybe not...

Comment (gjcarew) [reply lib/game_stats.rb:23]: @neb417 This could potentially be a "generic" calculation we could use in a module

Comment (neb417) [reply lib/game_stats.rb:23]: > 

Probably.

**Review by neb417 — APPROVED** [review]
Looked through and there are spots where module methods will work.

**Comment by neb417** [comment]
reviewed with Alex.

**Pull request merged**